### PR TITLE
Extend shapeInformation to set batch size for all inputs

### PR DESCRIPTION
**Pull request by tungld**
Option `--shapeInformation` currently allows setting static/dynamic dimensions for each model input. And, for each input, users need to set **all** dimensions, for example, if the first input is rank of 3, users need to pass a value say `0:-1x3x5` that must include 3 dimension sizes.

This patch extends `--shapeInformation` to allow partial setting of dimensions, counting from the outermost dimensions. For example, if users only want to set the first dimension of the first input to 5, users can use `0:5` and omit the remaining dimensions.

This new extension is useful when we set batch size for all inputs of a model. The batch size is often the first dimension (the outermost dimension), so we can use `-1:8` to set the batch size to 8 for all inputs.

### CODE REVIEW

File: src/Builder/ModelInputShaper.cpp [hunk 1]
```diff
@@ -55,13 +55,22 @@ void ModelInputShaper::setShapeInformation(
   if (!shapeInformation.empty()) {
     std::stringstream shapeInfoString(shapeInformation);
     std::string shapeString;
+    bool hasAllInputSetting = false;
     while (std::getline(shapeInfoString, shapeString, ',')) {
+      assert(!hasAllInputSetting &&
```
Comment (AlexandreEichenberger) [inline src/Builder/ModelInputShaper.cpp:60]: Would we possibly create an error as opposed to an assert? Better for user-fronting issues.

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM, great functionality.

Have you investigated if (for example when we have many model inputs having a batch size of -1), whether relying on the user to correlate them would help?

Or is dim analysis good enough to detect this on its own?

**Comment by tungld** [comment]
> Have you investigated if (for example when we have many model inputs having a batch size of -1), whether relying on the user to correlate them would help?

Yes, I have thought about that and need a way for users to express the relationship between dims.

File: src/Builder/ModelInputShaper.cpp (hunk 1 above)
Comment (tungld) [reply src/Builder/ModelInputShaper.cpp:60]: Yes, will change it to an error.

Comment (tungld) [reply src/Builder/ModelInputShaper.cpp]: Done. Thanks!

**Pull request merged**